app/controllers/admin/problems_controller.rb

```diff
@@ -70,6 +69,14 @@ def update
     end
   end
 
+  def toggle_runs_visible
+    @problem = Problem.find(params[:id])
```

Comment:
It seems to me that this `Problem.find` is duplicated across several actions. I think it can be moved into a private function like this:

```ruby
def problem
  @problem ||= Problem.find(params[:id)
end
```

and use `problem` instead of `@problem` in the controller actions that accept an `:id` parameter.